default-application-catalog/argocd-app.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: apps.kubermatic.k8c.io/v1
+kind: ApplicationDefinition
+metadata:
+  name: argocd
+spec:
+  description: Argo CD - Declarative, GitOps Continuous Delivery Tool for Kubernetes.
+  method: helm
+  versions:
+  - template:
+      source:
+        helm:
+          chartName: argo-cd
+          chartVersion: 5.5.12
+          url: https://argoproj.github.io/argo-helm
+    version: v2.4.14
+  # During the applicationdefinition creation time if the defaultValues is updated, then it is render for override during the application creation process from UI.
+  # By default, it is set empty.
+  # Assumption - LetsEncrypt clusterissuer named `letsencrypt-prod` is used for certificates management.
+  # Update the Domain name with appropriate value.
+  defaultValues:
```

Comment:
Does ArgoCD necessarily require an ingress to work? I am wondering if we could make our lives easier by just omitting the ingress and leaving this up to the end user

Reply:
Possible, fixed it!

Reply:
for now this is fine by me. I have added the topic to sig-app-mgmt for further discussion. Mainly because I am not 100% sure that we can assume customers to have LoadBalancer capabilities in their cluster. But this is for now fine and can easily be changed later on